sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/MessageFlux.java

```diff
@@ -0,0 +1,1045 @@
+    }
+
+    /**
+     * The callback invoked when next receiver is attached to the messaging entity from which this message-flux
+     * instance stream messages. There will be only one receiver at a time, and this callback delivers the reference
+     * to the function to disposition messages that arrives in the new receiver.
+     *
+     * @param updateDispositionFunc the function to disposition messages delivered by the current backing receiver.
+     */
+    void onNextUpdateDispositionFunction(BiFunction<String, DeliveryState, Mono<Void>> updateDispositionFunc) {
+        this.updateDispositionFunc = updateDispositionFunc;
+    }
+
+    /**
+     * The underlying consumer and producer extension of the message-flux operator. The consuming side processes events
+     * (about new receiver, terminal signals) from the upstream and events (messages, terminal signals) from
+     * the current receiver. The producing side publishes the messages to message-flux's downstream. The type has
+     * a recovery mechanism to obtain a new receiver from upstream upon the current receiver's termination.
+     * Recoveries happen underneath while the messages flow transparently downstream. The type can terminate downstream
+     * if the upstream terminates, the recovery path encounters a non-retriable error (i.e., the current receiver
+     * terminated with a non-retriable error), or recovery retries exhaust.
+     */
+    private static final class RecoverableReactorReceiver implements CoreSubscriber<AmqpReceiveLink>, Subscription {
+        // holds the current mediator that coordinates between a receiver and the recoverable-receiver.
+        private final MediatorHolder mediatorHolder = new MediatorHolder();
+        private final MessageFlux parent;
+        private final int prefetch;
+        private final CreditFlowMode creditFlowMode;
+        private final AmqpRetryPolicy retryPolicy;
+        private final ClientLogger logger;
+        private final AtomicInteger retryAttempts = new AtomicInteger();
+        private final CoreSubscriber<? super Message> messageSubscriber;
+        private Subscription upstream;
+        private volatile long requested;
+        @SuppressWarnings("rawtypes")
+        private static final AtomicLongFieldUpdater<RecoverableReactorReceiver> REQUESTED =
+            AtomicLongFieldUpdater.newUpdater(RecoverableReactorReceiver.class, "requested");
+        private volatile int wip;
+        @SuppressWarnings("rawtypes")
+        private static final AtomicIntegerFieldUpdater<RecoverableReactorReceiver> WIP =
+            AtomicIntegerFieldUpdater.newUpdater(RecoverableReactorReceiver.class, "wip");
+        private volatile boolean done;
+        private volatile boolean cancelled;
+        volatile Throwable error;
+        @SuppressWarnings("rawtypes")
+        static final AtomicReferenceFieldUpdater<RecoverableReactorReceiver, Throwable> ERROR =
```

Comment:
Curious why this is package private whereas the other ones are private?

Reply:
Ah, that was a miss, I’ll make it private.